1.2/spec.md

```diff
@@ -802,93 +802,116 @@ of grouped elements in the output and fail with an error message if they cannot
-document.
-
-The capability to generate specific properties is given by the prefix `ocrp_...`;
-the important properties are:
-
-## <dfn value for="ocr-capabilities">ocrp_lang</dfn>
+Metadata {#metadata}
+========
 
-Capable of generating `lang=` attributes
+The creator of the hOCR document can indicate the following information
+information using <{meta}> tags in the <{head}> section.
 
-## <dfn value for="ocr-capabilities">ocrp_dir</dfn>
+  : <dfn property>ocr-system</dfn>
+  :: Indicates software and version that generated the hOCR document
+  :: Every hOCR document *must* have one 'ocr-system' metadata field
 
-Capable of generating `dir=` attributes
+  : <dfn property>ocr-capabilities</dfn>
+  :: Features consumers of the hOCR document can expect
+  :: See [[#capabilities]] for possible values
+  :: Every hOCR document *must* have one 'ocr-capabilities' metadata field
 
-## <dfn value for="ocr-capabilities">ocrp_poly</dfn>
+  : <dfn property>ocr-number-of-pages</dfn>
+  :: The number of <{ocr_page}> in the document
 
-Capable of generating [polygonal bounds](#poly)
+  : <dfn property>ocr-langs</dfn>
+  :: Use [ISO 639-1](https://www.loc.gov/standards/iso639-2/php/code_list.php) codes
+  :: Value may be `unknown`
 
-## <dfn value for="ocr-capabilities">ocrp_font</dfn>
+  : <dfn property>ocr-scripts</dfn>
+  :: Use [ISO 15924](http://www.unicode.org/iso15924/codelists.html) letter codes
+  :: Value may be `unknown`
 
-Capable of generating font information (standard font information)
+Capabilities {#capabilities}
+------------
 
-## <dfn value for="ocr-capabilities">ocrp_nlp</dfn>
-
-Capable of generating [nlp confidences](#nlp)
-
-## `ocr_embeddedformat_<formatname>`
-
-The capability to generate other specific embedded formats is given by the
-prefix `ocr_embeddedformat_<formatname>`.
-
-## `ocr_<tag>_unordered`
-
-If an OCR engine represents a particular tag but cannot determine reading order
-for that tag, it must must specify a capability of `ocr_<tag>_unordered`.
+Any program generating files in this output format must indicate in the
+document metadata what kind of markup it is capable of generating. This
+includes listing the exact set of markup sections that the system could have
+generated, even if it did not actually generate them for the particular
+document.
 
 If a document lists a certain capabilities but no element or attribute is found
 that corresponds to that capability, users of the document may infer that the
 content is absent in the source document. If a capability is not listed, the
 corresponding element or attribute must not be present in the document.
 
+The capability to generate specific properties is given by the prefix `ocrp_...`;
+the important properties are:
+
+<dl dfn-for="ocr-capabilities">
+
+  : <dfn value>ocrp_lang</dfn>
+  :: Capable of generating `lang=` attributes
 
-# Metadata
+  : <dfn value for="ocr-capabilities">ocrp_dir</dfn>
```

Comment:
It's semantically equivalent to specify a `dfn-for=XYZ` attribute on a container element (in this case the definition list) to specifying `for=XYZ` on the `<dfn>` itself. https://tabatkins.github.io/bikeshed/#dfn-for